### PR TITLE
ButtonGroup: Fix button and icon button styling when tooltips are used on them

### PR DESCRIPTION
**Pull request by broccolinisoup**
Closes https://github.com/primer/react/issues/4129

#### Changed

Updated Button group CSS to make sure the icon buttons and buttons are rendered as expected when they have tooltips (both v1 and v2) on them

### Rollout strategy

- [x] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: src/Button/IconButton.features.stories.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import {HeartIcon} from '@primer/octicons-react'
 import React from 'react'
 import {IconButton} from '.'
+import {Tooltip} from '../drafts'
```
Comment (broccolinisoup) [inline src/Button/IconButton.features.stories.tsx:4]:
```suggestion
```

File: src/ButtonGroup/ButtonGroup.features.stories.tsx [hunk 2]
```diff
@@ -15,3 +16,47 @@ export const IconButtons = () => (
     <IconButton icon={DashIcon} aria-label="Subtract" />
   </ButtonGroup>
 )
+
+export const IconButtonsWithTooltip = () => (
```
Comment (broccolinisoup) [inline src/ButtonGroup/ButtonGroup.features.stories.tsx:20]: Happy to remove them after the review if we don't want to keep them

Comment (langermank) [reply src/ButtonGroup/ButtonGroup.features.stories.tsx:20]: I think we can leave em' :)

File: src/ButtonGroup/ButtonGroup.tsx [hunk 3]
```diff
@@ -8,28 +8,32 @@ const ButtonGroup = styled.div`
   vertical-align: middle;
   isolation: isolate;
 
-  && > * {
+  && > button,
```
Comment (langermank) [inline src/ButtonGroup/ButtonGroup.tsx:11]: @broccolinisoup is it possible these could be `a` tags as well? You could add a test story case for that and see.

**Review by langermank — COMMENTED** [review]
Just one small question, otherwise looks good 👍

File: src/ButtonGroup/ButtonGroup.tsx (hunk 3 above)
Comment (broccolinisoup) [reply src/ButtonGroup/ButtonGroup.tsx:11]: Oo great catch @langermank! Definitely, they can be `a` tags. I'll update the code.

**Comment by broccolinisoup** [comment]
ping @primer/reviewers

**Review by joshblack — APPROVED** [review]

**Pull request merged**